qa/workunits/ceph-helpers.sh

```diff
@@ -1048,6 +1048,44 @@ function test_repair() {
     ! TIMEOUT=1 repair 1.0 || return 1
     teardown $dir || return 1
 }
+#######################################################################
+
+##
+# Run scrub on **pgid** and wait until it completes. The repair
+# function will fail if repair does not complete within $TIMEOUT
+# seconds. The scrub is complete whenever the
+# **get_last_scrub_stamp** function reports a timestamp different from
+# the one stored before starting the scrub.
+#
+# @param pgid the id of the PG
+# @return 0 on success, 1 on error
+#
+function pg_scrub() {
+    local pgid=$1
+    local last_scrub=$(get_last_scrub_stamp $pgid)
+
+    ceph pg scrub $pgid
+    for ((i=0; i < $TIMEOUT; i++)); do
+        if test "$last_scrub" != "$(get_last_scrub_stamp $pgid)" ; then
+            echo "pg $pgid scrub finish."
+            return 0
+        fi
+        sleep 1
```

Comment:
Doing that will put lots of duplicate stuff in the log.  How about this in pg_scrub()?

ceph -w &
ceph pg scrub $pgid
for ....
...
    kill %%
    return 0
done
kill %%
return 1